src/core/algorithms/dd/dd_verifier/dd_verifier.cpp

```diff
@@ -0,0 +1,211 @@
+        model::ColumnIndex column_index =
+                typed_relation_->GetSchema()->GetColumn(constraint.column_name)->GetIndex();
+        columns.push_back(column_index);
+    }
+    auto curr_constraint = dd_.left.cbegin();
+    for (std::size_t i = 0; i < num_rows_; i++) {
+        for (std::size_t j = i + 1; j < num_rows_; j++) {
+            if (auto const dif = CalculateDistance(columns[0], {i, j});
+                curr_constraint->constraint.Contains(dif)) {
+                result.emplace_back(i, j);
+            }
+        }
+    }
+    ++curr_constraint;
+    for (std::size_t i = 1; i < columns.size(); i++) {
+        std::vector<std::pair<int, int>> new_result;
+        for (std::size_t j = 0; j < result.size(); j++) {
+            if (double const dif = CalculateDistance(columns[i], result[j]);
+                curr_constraint->constraint.Contains(dif)) {
+                new_result.emplace_back(result[j]);
+            }
+        }
+        result = std::move(new_result);
+        ++curr_constraint;
+    }
+    return result;
+}
+
+void DDVerifier::LoadDataInternal() {
+    typed_relation_ = model::ColumnLayoutTypedRelationData::CreateFrom(*input_table_, false);
+}
+
+void DDVerifier::MakeExecuteOptsAvailable() {
+    using namespace config::names;
+    MakeOptionsAvailable({kDDString});
+}
+
+void DDVerifier::CheckCorrectnessDd() const {
+    for (auto const &constraint : dd_.left) {
+        assert(constraint.constraint.upper_bound >= constraint.constraint.lower_bound);
+        assert(constraint.constraint.lower_bound >= 0);
+        assert(constraint.constraint.upper_bound >= 0);
+    }
+    for (auto const &constraint : dd_.right) {
+        assert(constraint.constraint.upper_bound >= constraint.constraint.lower_bound);
+        assert(constraint.constraint.lower_bound >= 0);
+        assert(constraint.constraint.upper_bound >= 0);
+    }
+}
+
+unsigned long long DDVerifier::ExecuteInternal() {
+    std::vector<model::ColumnIndex> columns;
+    for (auto const &constraint : dd_.left) {
+        model::ColumnIndex column_index =
+                typed_relation_->GetSchema()->GetColumn(constraint.column_name)->GetIndex();
+        columns.push_back(column_index);
+    }
+    for (auto const &constraint : dd_.right) {
+        model::ColumnIndex column_index =
+                typed_relation_->GetSchema()->GetColumn(constraint.column_name)->GetIndex();
+        columns.push_back(column_index);
+    }
+    for (auto const col : columns) {
+        assert(IsColumnMetrizable(col));
+    }
+    CheckCorrectnessDd();
+    num_rows_ = typed_relation_->GetNumRows();
+    num_columns_ = typed_relation_->GetNumColumns();
+
+    auto start_time = std::chrono::system_clock::now();
+
+    VerifyDD();
+
+    PrintStatistics();
+
+    auto elapsed_milliseconds = std::chrono::duration_cast<std::chrono::milliseconds>(
+            std::chrono::system_clock::now() - start_time);
+
+    return elapsed_milliseconds.count();
+}
```

Comment:
nit: you could consider moving all code from `ExecuteInternal` to `VerifyDD`, and do this:
```cpp
unsigned long long DDVerifier::ExecuteInternal() {
    return util::TimedInvoke(&DDVerifier::VerifyDD, this);
}
```